### PR TITLE
Added progress logging to MWM export process.

### PR DESCRIPTION
**Pull request by strump**
MWM export process takes couple of hours for every country/region but it has no progress indicator. I found the longest loop in `GenerateFinalFeatures` function and added logging:

**Before:**
```
[2021-08-04 11:41:45,542] INFO stages Stage Index: start ...
LOG TID(1) DEBUG  4.2051e-05 platform/platform_mac.mm:111 Platform Resources Directory: ./data/
LOG TID(1) DEBUG 0.000113293 platform/platform_mac.mm:112 Platform Writable Directory: ./data/
LOG TID(1) DEBUG 0.000139213 platform/platform_mac.mm:113 Platform Tmp Directory: /var/folders/br/rhx65r4x4d56dthrjd9w9s400000gn/T//
LOG TID(1) DEBUG 0.000158241 platform/platform_mac.mm:114 Platform Settings Directory: ./data/
LOG TID(1) DEBUG 0.000308233 indexer/classificator_loader.cpp:47 Load() Reading of classificator started
[2021-08-04 11:41:45,600] INFO gen_tool Run generator tool [generator_tool version 1627653421 ab6697d982ed67c4ce9ad8384569ea744733b929-dirty Debug build (NDEBUG not #defined)]: /Users/.../omim-build-debug/generator_tool --threads_count=12 --data_path=/Users/.../osm_data/mwm/2021_08_04__11_10_41/210804 --intermediate_data_path=/Users/.../osm_data/mwm/2021_08_04__11_10_41/intermediate_data --cache_path=/Users/.../osm_data/mwm/2021_08_04__11_10_41/intermediate_data --user_resource_path=/Users/.../organicmaps/data --node_storage=map --planet_version=1628064641 --generate_geometry=true --generate_index=true --output=Ukraine_Kyiv Oblast --generate_search_index=true 
LOG TID(1) DEBUG    0.315908 indexer/classificator_loader.cpp:69 Load() Reading of classificator finished
LOG TID(1) INFO     0.315971 generator_tool/generator_tool.cpp:360 operator()() Generating result features for Ukraine_Kyiv Oblast
LOG TID(1) WARNING   16.4879 generator/region_meta.cpp:68 ReadRegionDataImpl() File leap_speeds.json not found.
LOG TID(1) WARNING   16.4898 generator/region_meta.cpp:68 ReadRegionDataImpl() File leap_speeds.json not found.
LOG TID(1) WARNING   16.9949 generator/geometry_holder.hpp:205 WriteOuterTriangles() GeometryHolder: Large feature Osm Relation 1017128 with 11069 points
LOG TID(1) WARNING   20.2317 generator/geometry_holder.hpp:205 WriteOuterTriangles() GeometryHolder: Large feature Osm Relation 3173129 with 28500 points
LOG TID(1) WARNING    28.327 generator/geometry_holder.hpp:205 WriteOuterTriangles() GeometryHolder: Large feature Osm Relation 1605938 with 40172 points
LOG TID(1) WARNING   80966.4 generator/borders.cpp:172 GetBordersRect() File with borders does not exist: /Users/.../osm_data/mwm/2021_08_04__11_10_41/210804/borders/Ukraine_Kyiv Oblast.poly
LOG TID(1) INFO      80969.7 generator_tool/generator_tool.cpp:364 operator()() Generating offsets table for /Users/.../osm_data/mwm/2021_08_04__11_10_41/210804/Ukraine_Kyiv Oblast.mwm
LOG TID(1) INFO      80970.8 indexer/features_offsets_table.cpp:93 Save() Saving features offsets table to  /Users/.../osm_data/mwm/2021_08_04__11_10_41/210804/Ukraine_Kyiv Oblast.mwm.offsets
...
```
Between "Generating result features for " and "Generating offsets table" lines there's not many message.

**After:**
```
[2021-08-05 14:23:04,718] INFO stages Stage Index: start ...
LOG TID(1) DEBUG  3.4314e-05 platform/platform_mac.mm:111 Platform Resources Directory: ./data/
LOG TID(1) DEBUG  9.8019e-05 platform/platform_mac.mm:112 Platform Writable Directory: ./data/
LOG TID(1) DEBUG 0.000120639 platform/platform_mac.mm:113 Platform Tmp Directory: /var/folders/br/rhx65r4x4d56dthrjd9w9s400000gn/T//
LOG TID(1) DEBUG 0.000137061 platform/platform_mac.mm:114 Platform Settings Directory: ./data/
LOG TID(1) DEBUG 0.000276889 indexer/classificator_loader.cpp:47 Load() Reading of classificator started
[2021-08-05 14:23:04,856] INFO gen_tool Run generator tool [generator_tool version 1628066902 0659467a622160b0059a95e6925b13c16430044f Debug build (NDEBUG not #defined)]: /Users/.../omim-build-debug/generator_tool --threads_count=12 --data_path=/Users/.../osm_data/mwm/2021_08_05__13_42_29/210805 --intermediate_data_path=/Users/.../osm_data/mwm/2021_08_05__13_42_29/intermediate_data --cache_path=/Users/.../osm_data/mwm/2021_08_05__13_42_29/intermediate_data --user_resource_path=/Users/.../organicmaps/data --node_storage=map --planet_version=1628160149 --generate_geometry=true --generate_index=true --output=Ukraine_Kyiv Oblast --generate_search_index=true 
LOG TID(1) DEBUG    0.303435 indexer/classificator_loader.cpp:69 Load() Reading of classificator finished
LOG TID(1) INFO     0.303502 generator_tool/generator_tool.cpp:360 operator()() Generating result features for Ukraine_Kyiv Oblast
LOG TID(1) WARNING   16.8296 generator/region_meta.cpp:68 ReadRegionDataImpl() File leap_speeds.json not found.
LOG TID(1) WARNING   16.8333 generator/region_meta.cpp:68 ReadRegionDataImpl() File leap_speeds.json not found.
LOG TID(1) INFO      16.9291 generator/feature_sorter.cpp:376 GenerateFinalFeatures() 872 of 871971 points ( 0 %) exported to MWM
LOG TID(1) WARNING    17.389 generator/geometry_holder.hpp:205 WriteOuterTriangles() GeometryHolder: Large feature Osm Relation 1017128 with 11069 points
LOG TID(1) WARNING   20.8618 generator/geometry_holder.hpp:205 WriteOuterTriangles() GeometryHolder: Large feature Osm Relation 3173129 with 28500 points
LOG TID(1) WARNING   30.1534 generator/geometry_holder.hpp:205 WriteOuterTriangles() GeometryHolder: Large feature Osm Relation 1605938 with 40172 points
LOG TID(1) INFO      51.2008 generator/feature_sorter.cpp:382 GenerateFinalFeatures() 1742 of 871971 points ( 0 %) exported to MWM
LOG TID(1) INFO      56.5206 generator/feature_sorter.cpp:382 GenerateFinalFeatures() 2613 of 871971 points ( 0 %) exported to MWM
LOG TID(1) INFO      68.1476 generator/feature_sorter.cpp:382 GenerateFinalFeatures() 3484 of 871971 points ( 0 %) exported to MWM
LOG TID(1) INFO      72.6905 generator/feature_sorter.cpp:382 GenerateFinalFeatures() 4355 of 871971 points ( 0 %) exported to MWM
LOG TID(1) INFO      80.1539 generator/feature_sorter.cpp:382 GenerateFinalFeatures() 5226 of 871971 points ( 0 %) exported to MWM
LOG TID(1) INFO       129.78 generator/feature_sorter.cpp:382 GenerateFinalFeatures() 6097 of 871971 points ( 0 %) exported to MWM
LOG TID(1) INFO      140.145 generator/feature_sorter.cpp:382 GenerateFinalFeatures() 6968 of 871971 points ( 0 %) exported to MWM
LOG TID(1) INFO      166.742 generator/feature_sorter.cpp:382 GenerateFinalFeatures() 7839 of 871971 points ( 0 %) exported to MWM
LOG TID(1) INFO      175.121 generator/feature_sorter.cpp:382 GenerateFinalFeatures() 8710 of 871971 points ( 0 %) exported to MWM
LOG TID(1) INFO      184.936 generator/feature_sorter.cpp:382 GenerateFinalFeatures() 9581 of 871971 points ( 1 %) exported to MWM
LOG TID(1) INFO      191.742 generator/feature_sorter.cpp:382 GenerateFinalFeatures() 10452 of 871971 points ( 1 %) exported to MWM
LOG TID(1) INFO      193.413 generator/feature_sorter.cpp:382 GenerateFinalFeatures() 11323 of 871971 points ( 1 %) exported to MWM
LOG TID(1) INFO      199.846 generator/feature_sorter.cpp:382 GenerateFinalFeatures() 12194 of 871971 points ( 1 %) exported to MWM
LOG TID(1) INFO      204.855 generator/feature_sorter.cpp:382 GenerateFinalFeatures() 13065 of 871971 points ( 1 %) exported to MWM
...
```

Print 10 lines per 1 percent of progress.

### CODE REVIEW

File: generator/feature_sorter.cpp [hunk 1]
```diff
@@ -360,14 +360,21 @@ bool GenerateFinalFeatures(feature::GenerateInfo const & info, string const & na
       // We cannot remove it in ~FeaturesCollector2(), we need to remove it in SCOPE_GUARD.
       SCOPE_GUARD(_, [&]() { Platform::RemoveFileIfExists(info.GetTargetFileName(name, FEATURES_FILE_TAG)); });
       FeaturesCollector2 collector(name, info, header, regionData, info.m_versionDate);
-      for (auto const & point : midPoints.GetVector())
+      auto points = midPoints.GetVector();
```
Comment (vng) [inline generator/feature_sorter.cpp:363]: Use auto const & points = ... to avoid vector copy creation.

File: generator/feature_sorter.cpp [hunk 2]
```diff
@@ -360,14 +360,21 @@ bool GenerateFinalFeatures(feature::GenerateInfo const & info, string const & na
...
       {
         ReaderSource<FileReader> src(reader);
         src.Skip(point.second);
 
         FeatureBuilder fb;
         ReadFromSourceRawFormat(src, fb);
         collector(fb);
+        if ( (++processedPoints) % promillePoints == 0) {
+          LOG(LINFO, (processedPoints, "of", totalPoints, "points (", (processedPoints*100/totalPoints) ,"%) exported to MWM"));
+        }
```
Comment (vng) [inline generator/feature_sorter.cpp:377]: Without space and braces:
```
if (++processedPoints % promillePoints == 0)
  LOG(LINFO, (processedPoints, "of", totalPoints, "points (", (processedPoints*100/totalPoints) ,"%) exported to MWM"));
```

**Review by vng — COMMENTED** [review]
I don't mind about this patch, but this is definitely not the longest routine.
Building routing index is way much longer..

Also better to use Release generator_tool, except if you are debugging for a bug.

**Comment by strump** [comment]
Release version is lightning fast! Looks like progress bar is not needed for `GenerateFinalFeatures`

**Pull request closed without merging**